packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -404,14 +424,12 @@ options:
     final List<Error> errors = <Error>[];
     final List<Type> apis = <Type>[];
     if (options.objcHeaderOut != null) {
-      options.objcOptions.header = basename(options.objcHeaderOut);
-    }
-    if (options.javaOut != null) {
-      options.javaOptions.className = basenameWithoutExtension(options.javaOut);
+      options.objcOptions?.header = basename(options.objcHeaderOut!);
     }
```

Comment:
Here's the code that creates the className, it looks like it might have been lost.

Reply:
Not lost, accounted for.